src/app/static/src/app/components/forms/CustomForm.vue

```diff
@@ -0,0 +1,60 @@
+<template>
+    <b-form :ref="id" class="custom-form">
+        <div v-for="section in form.controlSections">
+            <h3>{{section.label}}</h3>
+            <p class="text-muted">{{section.description}}</p>
+            <control-group v-for="group in section.controlGroups" :group="group"></control-group>
+        </div>
+        <button v-if="includeSubmitButton" class="btn btn-red" v-on:click="submit">{{submitText}}</button>
+    </b-form>
+</template>
+
+<script lang="ts">
+
+    import Vue from "vue";
+    import {BForm} from "bootstrap-vue";
+    import {formMeta} from "./fakeFormMeta";
+    import ControlGroup from "./ControlGroup.vue";
+    import {Dictionary} from "vuex";
+
+    export default Vue.extend({
+        name: "CustomForm",
+        props: {
+            id: {
+                type: String,
+                default: Math.random().toString(36).slice(-5)
+            },
+            submitText: {
+                type: String,
+                default: "Submit"
+            },
+            includeSubmitButton: {
+                type: Boolean,
+                default: true
+            }
+        },
+        data() {
+            return {
+                form: formMeta
+            }
+        },
+        components: {
+            BForm,
+            ControlGroup
+        },
+        methods: {
+            submit(e: Event) {
```

Comment:
So if includeFormButton = false (and so e is falsy), that means that the submit method will be called from outside the form (as is done when integrated into the ModelComponent)? So in that case, it might make more sense to return the data from this method rather than to emit the event?

Reply:
Ooh yes good idea. And no reason for it not to do both, so the user can choose how to consume it.